src/mlpack/methods/gmm/gmm.cpp

```diff
@@ -66,17 +69,56 @@ double GMM::LogProbability(const arma::vec& observation) const
   return sum;
 }
 
+/**
+ * Return the log probability of the given observation GMM matrix.
+ *
+ * @param x Observation Matrix to compute log-probabilty.
+ * @param logProbs Stores the value of log-probability for x.
+ */
+void GMM::LogProbability(const arma::mat& x, arma::vec& logProbs) const
+{
+  // Sum the probability for each Gaussian in our mixture (and we have to
+  // multiply by the prior for each Gaussian too).
+  logProbs.set_size(x.n_cols);
+  double sum;
+  for (size_t j=0; j < x.n_cols; j++)
+  {
+    sum = -std::numeric_limits<double>::infinity();
+    for (size_t i = 0; i < gaussians; i++)
+      sum = math::LogAdd(sum, log(weights[i]) +
+          dists[i].LogProbability(x.unsafe_col(j)));
+    logProbs(j) = sum;
+  }
+}
+
 /**
  * Return the probability of the given observation being from this GMM.
+ *
+ * @param observation Observation Matrix to compute probabilty.
  */
 double GMM::Probability(const arma::vec& observation) const
 {
   return exp(LogProbability(observation));
 }
 
+/**
+ * Return the probability of the given observation GMM matrix.
+ *
+ * @param x Observation Matrix to compute probabilty.
+ * @param probs Stores the value of probability for x.
+ */
+void GMM::Probability(const arma::mat& x, arma::vec& probs) const
+{
+  LogProbability(x, probs);
+  probs = exp(probs);
+}
+
 /**
  * Return the log probability of the given observation being from the given
  * component in the mixture.
+ *
+ * @param observation Observation Matrix to compute log-probabilty.
```

Comment:
Should this be `Observation to compute log-probability for.`?  This takes a individual vector, not a matrix.